### ISSUE TITLE
Enquiry about sods speed

### ISSUE THREAD

**Issue by yanggis** [msantos/sods#3]
Hi, firstly, thanks for sharing the great app. May I ask if you test the speed with sods tunnel, for example, max downloading speed?
I've tried iodine for DNS tunnel; iodine is stable, but the speed is a bit unsatisfied.

**Comment by msantos** [comment]
I haven't done a speed test in a while but the speed should be similar to iodine. The main factors will be:

* the rate packets can be sent via the intermediary DNS servers

   For example, if I go directly from the sods client to the sods server, it is fast enough not to notice any lag in a terminal session. When going through an intermediary, depending on the DNS server, the lag can be quite noticeable.

* the amount of data sent in each packet

  NULL DNS records can have the largest payload but aren't widely supported.

* the amount of data

  If I remember correctly, iodine tunnels the data, sending either the whole frame or the IP packet. sods embeds the application data only.

So tl;dr not very fast. Thanks for the question @yanggis !

**Comment by yanggis** [comment]
Dear msantos, 
Thanks for all clarifications. In fact, I've made an investigation, so wanna to move from iodine to sods. Now, I just use iodine for data transfer via a DNS server, but the speed is only up to 150k/s. 
May I know config sods so that sods can go directly from the sods client to the sods server? I just checked the config file, no tips on how to make this configuration.

**Comment by msantos** [comment]
You can specify the nameserver using the "-r" flag. For example, if your sods server was 8.8.8.8:
~~~
ssh -o ProxyCommand="sdt -r 8.8.8.8 x.o.example.com" 127.0.1.100
~~~

**Comment by yanggis** [comment]
Thanks for your fast response. I'll try and let you know the results.

**Comment by 1937lilizhou** [comment]
hi 
msantos，Could you tell me how to use the sods?What do  i need to do something when i use it? I run it by readme, but i always fail.Do I need to build a DNS server called a.example.com  in the server of sods?
